overlord/overlord_test.go

```diff
@@ -596,20 +616,25 @@ func (ovs *overlordSuite) TestEnsureLoopPruneRunsMultipleTimes(c *C) {
 	c.Check(st.Changes(), HasLen, 2)
 	st.Unlock()
 
+	w, restoreTicker := fakePruneTicker()
+	defer restoreTicker()
+
 	// start the loop that runs the prune ticker
 	o.Loop()
 
-	// ensure the first change is pruned
-	time.Sleep(1500 * time.Millisecond)
-	st.Lock()
-	c.Check(st.Changes(), HasLen, 1)
-	st.Unlock()
+	// this needs to be more than pruneWait=5ms mocked above
+	time.Sleep(6 * time.Millisecond)
+	w.tick(2)
```

Comment:
Why do we tick twice here (instead of one or three or ten)? I looked at this a bit because I'm puzzled that 1 does not work. I wonder if this is still racy. When we do the w.tick(1) here we sent the time value to the channel but there is no synchronization that the value is received and processed. It seems like "2" happens to trigger that the value is actually received on the other side. If that is correct probably worth documenting.

Reply:
This is needed for extra Ensure, so that tasks/changes are updated properly for the test.

Reply:
It's not really important (sorry, this is a bit of OCD on my part) but I think it's really the async nature of channels that we see here. If I change the code to:
```
        time.Sleep(6 * time.Millisecond)
	w.tick(1)
	time.Sleep(1)
```
it works and I think it does because the sleep wakes up the other go-routine that waits for the channel. Anyway, not really that important, just that it bugged me that I did not fully understood it :)

Reply:
that 2nd sleep is very suspicious though